freqtrade/strategy/interface.py

```diff
@@ -108,6 +108,7 @@ class IStrategy(ABC, HyperStrategyMixin):
 
     # Position adjustment is disabled by default
     position_adjustment_enable: bool = False
+    max_buy_position_adjustment: int = 1
```

Comment:
the use of this (and how it's named) will only depend on the strategy, and how the strategy implements it (the documentation is just a sample, not the default behavior). We could have different samples, using different calculation methods (also ignoring this completely).

So by adding this as configuration element (suggesting it is applied internally), can be confusing and potentially dangerous - if you assume your setting comes from the configuration, but the strategy you're using is really using different naming (or calculates differently altogether).

Reply:
I'll make it to behave similar to max_open_trades. -1 for unlimited adjust buy (which is default), which can either be set to another value on strat file if needed, or let the coder create their own logic.

if the max limit is set to a positive integer, the bot will check internally for number of buys. If it's still below the limit, then adjust_position won't be called. So this snippet from the docs won't be needed anymore

```filled_buys = trade.select_filled_orders('buy')
        count_of_buys = len(filled_buys)
        
        # Allow up to 3 additional increasingly larger buys (4 in total)
        # Initial buy is 1x
        # If that falls to -5% profit, we buy 1.25x more, average profit should increase to roughly -2.2%
        # If that falls down to -5% again, we buy 1.5x more
        # If that falls once again down to -5%, we buy 1.75x more
        # Total stake for this trade would be 1 + 1.25 + 1.5 + 1.75 = 5.5x of the initial allowed stake.
        # That is why max_dca_multiplier is 5.5
        # Hope you have a deep wallet!
        if 0 < count_of_buys <= self.max_buy_position_adjustment:```